src/psyclone/dynamo0p3.py

```diff
@@ -7141,7 +7141,10 @@ def lower_to_language_level(self):
             pos = self.position
             parent = self.parent
             self.detach()
-            for child in self.loop_body.pop_all_children():
+            all_children_reverse = self.loop_body.pop_all_children()[::-1]
+            # Attach the children starting with the last, which
+            # preserves the origina order of the children.
```

Comment:
"original"

Reply:
Done

Reply:
(And I finally changed the default setting of my editor to always enable spell checking - I hope that helps at least with the embarrassing typos)